sklearn/manifold/_spectral_embedding.py

```diff
@@ -364,8 +367,9 @@ def spectral_embedding(
             # We increase the number of eigenvectors requested, as lobpcg
             # doesn't behave well in low dimension and create initial
             # approximation X to eigenvectors
-            X = random_state.rand(laplacian.shape[0], n_components + 1)
+            X = random_state.randn(laplacian.shape[0], n_components + 1)
```

Comment:
Ideally this would have been fixed in a separate PR with its own changelog entry and potentially a new non-regression test (if possible).

If it's too complicated to disentangle the two fixes, then at least please add a new changelog entry to document this fix that also impacts the solver when running with `np.float64` data.

Reply:
not complicated to disentangle the two fixes, so I remove it from here and make a separate PR after this one is merged

Reply:
I have created a separate issue https://github.com/scikit-learn/scikit-learn/issues/21555 to address this in a separate PR